great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -4787,21 +4783,6 @@ def include_rendered_content(self) -> IncludeRenderedContentConfig:
     def notebooks(self) -> NotebookConfig:
         return self.variables.notebooks  # type: ignore[return-value]
 
-    @property
-    def datasources(
-        self,
-    ) -> Dict[str, Union[LegacyDatasource, BaseDatasource, FluentDatasource]]:
-        """A single holder for all Datasources in this context"""
-        return self._cached_datasources
-
-    @property
-    def fluent_datasources(self) -> Dict[str, FluentDatasource]:
-        return {
-            name: ds
-            for (name, ds) in self.datasources.items()
-            if isinstance(ds, FluentDatasource)
-        }
```

Comment:
Thinking about it more I do think we should do this by making a `MutableMapping` or custom `UserDict`
whose `__getitem__` method uses `self._datasource_store.retrieve_by_name(name)`.
Then the logic for accessing it doesn't need to change it all.

https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes
https://realpython.com/inherit-python-dict/#subclassing-userdict-from-collections
```python
class StoreDict(UserDict):

   def __item__(self, store)
        self._store = store
        super().__init__()

    def __getitem__(self, key: str):
        return self._store.retrieve_by_name(name)

    ...

```



And we can keep these public properties.